src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -525,17 +502,25 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             }
 
             var analyzedNullabilitiesMap = analyzedNullabilities.ToImmutable();
-            var updatedSymbolsMap = updatedSymbols.ToImmutable();
-            newSnapshots = newSnapshotBuilder?.ToManagerAndFree();
+            newSnapshots = newSnapshotBuilder.ToManagerAndFree();
 
 #if DEBUG
             if (binder.Compilation.NullableSemanticAnalysisEnabled)
             {
-                DebugVerifier.Verify(analyzedNullabilitiesMap, updatedSymbolsMap, newSnapshots, node);
+                DebugVerifier.Verify(analyzedNullabilitiesMap, newSnapshots, node);
             }
 #endif
 
-            return new NullabilityRewriter(analyzedNullabilitiesMap, updatedSymbolsMap).Visit(node);
+            return Rewrite(analyzedNullabilitiesMap, newSnapshots, node, out remappedSymbols);
+        }
+
+        private static BoundNode Rewrite(ImmutableDictionary<BoundExpression, (NullabilityInfo, TypeSymbol)> updatedNullabilities, SnapshotManager snapshotManager, BoundNode node, out ImmutableDictionary<Symbol, Symbol> remappedSymbols)
+        {
+            var remappedSymbolsBuilder = ImmutableDictionary.CreateBuilder<Symbol, Symbol>(SymbolEqualityComparer.ConsiderEverything, SymbolEqualityComparer.ConsiderEverything);
```

Comment:
>SymbolEqualityComparer.ConsiderEverything [](start = 91, length = 41)

Is this comparer based on ISymbol?

Reply:
Yes, but as far as I know we do not have an implementation of `IComparer<Symbol>`.

---
In reply to: [329722219](https://github.com/dotnet/roslyn/pull/38769#discussion_r329722219) [](ancestors = 329722219)